keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -478,46 +487,84 @@ Because Windows privileged containers will work much differently than Linux priv
+    this volume will be mounted to `c:\C\{container-id}`.
+  - Additional volumes mounts specified for `hostProcess` containers will be mounted to `c:\C\{container-id}\{mount-destination}`
+    - ex: a volume with a mountPath of `/var/run/secrets/token` for a container with id `1234` can be accessed at `c:\C\1235\var\run\secrets\token`
+  - An environment variable `$CONTAINER_SANDBOX_MOUNT_POINT` will be set to the path where the container volume is mounted (`c:\c\{container-id}`) to access content.
+    - This environment variable can be used inside the Pod manifest / command line / args for containers.
+
+A recording of the behaviors differces from a SIG-Windows community meeting can be found [here](https://youtu.be/8GeZKXgvkdY?t=309).
+Note -  In the recording it was mentioned that this functionally might not be supported on WS2019.
+    This functionality will be avaible in WS2019 but will require an OS patch (ETA: July 2022).
+
+Additionally the following will be true for either volume mount behavior:
+
+- Named Pipe mounts will **not** be supported.
+    Instead named pipes should be accessed via their path on the host (\\\\.\\pipe\\*).
+    The following error will be returned if `hostProcess` containers attempt to use name pipe mounts -
+    https://github.com/microsoft/hcsshim/blob/358f05d43d423310a265d006700ee81eb85725ed/internal/jobcontainers/mounts.go#L40.
+- Unix domain sockets mounts also not not be supported for `hostProces` containers.
+    Unix domain sockets can be accessed via their paths on the host like named pipes.
+- Mounting directories from the host OS into `hostProcess` containers will work just like with normal containers but this is not recommend.
+    Instead workloads should access the host OS's file-system as if not being run in a container.
+  - All other volume types supported for normal containers on Windows will work with `hostProcess` containers.
+- `HostProcess` Containers will have full access to the host file-system (unless restricted by filed-based ACLs and the run_as_username used to start the container).
+- There will be no `chroot` equivalent.
+
+##### Compatibility
+
+During the alpha/beta implementations of this feature only **Symlink** volume mount behavior was implmented.
+This implemention did unlock a lot of critical use cases for managing Windows nodes in Kubernets clusters but did have some usability issues
+(such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig not working as expected).
+
+The **bind mount** volume mount behavior gives full access to the host OS's filesystem (an explicit goal of this enhancement) and addreses the usability issues with the initial approach.
+This approach requires the use of Windows OS APIs that were not present in Windows Server 2019 during alpha/beta implmentations of this feature.
+These APIs *will* be available in WS2019 beginning in July 2022 with the monthly OS security patches.
+Containerd v1.7+ will be required for this behavior.
```

Comment:
what is the ETA for containerd 1.7?

Reply:
There are not currently any other relevant container runtimes.
`CRI-O` does not currently have any support on windows and all forms of Docker (Moby, Docker EE, cri-dockerd) are still being using on an older versions of the [HCS](https://docs.microsoft.com/en-us/virtualization/community/team-blog/2017/20170127-introducing-the-host-compute-service-hcs) APIs which cannot support host-process containers.

Reply:
The ETA for containerd 1.7 is Oct 2022 (last I heard).

This definitely puts this KEP in an interesting position.
The container runtime (containerd) is responsible for setting up the volume mounts for containers and this tightly couples the volume mount behaviors used for the nodes.

Currently containerd v1.6 will always use the `symlink` volume mount behavior (which has some considerable drawbacks) and having containerd v1.6 support the new `bind` volume mount behavior isn't likely due to the changes being potentially destabilizing. 

This also means that users are free to (and will likely be encouraged to) run with containerd v1.7 in older K8s clusters (prior to v1.25) which means they would get the new/desired `bind` mount behavior.

My opinion is that we put out lots of documentation so users can expect a change in behavior when switching upgrading containerd v1.6 to v1.7. 
If anyone has any thoughts on how we can make this transition easier for users without needing to support the `symlink` volume mount behavior indefinitely I'd be happy to hear it.

Reply:
> will the mount / path approach described here be documented as a requirement for supporting hostProcess containers, and an e2e (ideally conformance) test be written to exercise it?

We have e2e tests that exercises the existing `symlink` behavior today (they were added in v1.23) and had added writing new e2e tests that exercise the new `bind` behavior as part of GA graduation.

There currently is no way to mark Windows specific tests as conformance tests (more details in https://github.com/kubernetes/kubernetes/issues/96639)

We'll make sure to add these requriements as part of https://github.com/kubernetes/enhancements/issues/2578

Reply:
Since the symlink behavior and container reliance on $CONTAINER_SANDBOX_MOUNT_POINT is not what we want long-term, and indeed is almost deprecated-on-arrival since bind mounts are on the horizon, I think it's questionable to mark this feature GA before the bind mount functionality is available.

I really don't want containers taking dependencies on $CONTAINER_SANDBOX_MOUNT_POINT to use a supposedly GA feature when we know that is going away ~soon

Reply:
Agreed. We'll keep this in beta for one more release.